tests/serverpod_test_server/lib/src/generated/scopes/scope_server_only_field_child.dart

```diff
@@ -0,0 +1,117 @@
+/* AUTOMATICALLY GENERATED CODE DO NOT MODIFY */
+/*   To generate run: "serverpod generate"    */
+
+// ignore_for_file: implementation_imports
+// ignore_for_file: library_private_types_in_public_api
+// ignore_for_file: non_constant_identifier_names
+// ignore_for_file: public_member_api_docs
+// ignore_for_file: type_literal_in_constant_pattern
+// ignore_for_file: use_super_parameters
+
+// ignore_for_file: no_leading_underscores_for_library_prefixes
+import '../protocol.dart' as _i1;
+import 'package:serverpod/serverpod.dart' as _i2;
+
+abstract class ScopeServerOnlyFieldChild extends _i1.ScopeServerOnlyField
+    implements _i2.SerializableModel, _i2.ProtocolSerialization {
+  ScopeServerOnlyFieldChild._({
+    super.allScope,
+    super.serverOnlyScope,
+    super.nested,
+    required this.childFoo,
+  });
+
+  factory ScopeServerOnlyFieldChild({
+    _i1.Types? allScope,
+    _i1.Types? serverOnlyScope,
+    _i1.ScopeServerOnlyField? nested,
+    required String childFoo,
+  }) = _ScopeServerOnlyFieldChildImpl;
+
+  factory ScopeServerOnlyFieldChild.fromJson(
+      Map<String, dynamic> jsonSerialization) {
+    return ScopeServerOnlyFieldChild(
+      allScope: jsonSerialization['allScope'] == null
+          ? null
+          : _i1.Types.fromJson(
+              (jsonSerialization['allScope'] as Map<String, dynamic>)),
+      serverOnlyScope: jsonSerialization['serverOnlyScope'] == null
+          ? null
+          : _i1.Types.fromJson(
+              (jsonSerialization['serverOnlyScope'] as Map<String, dynamic>)),
+      nested: jsonSerialization['nested'] == null
+          ? null
+          : _i1.ScopeServerOnlyField.fromJson(
+              (jsonSerialization['nested'] as Map<String, dynamic>)),
+      childFoo: jsonSerialization['childFoo'] as String,
+    );
+  }
+
+  String childFoo;
+
+  @override
+  ScopeServerOnlyFieldChild copyWith({
+    Object? allScope,
+    Object? serverOnlyScope,
+    Object? nested,
+    String? childFoo,
+  });
```

Comment:
Question: Shouldn't a typed interface be used here? Isn't it only inherited classes that do not have support for type copy with interfaces?